src/jit/codegencommon.cpp

```diff
@@ -8519,6 +8519,13 @@ void                CodeGen::genFnProlog()
         genPrologPadForReJit();
         getEmitter()->emitMarkPrologEnd();
     }
+    
+#if defined(FEATURE_UNIX_AMD64_STRUCT_PASSING) && defined(FEATURE_SIMD)
+    // TODO-Cleanup: This logic can be implemented in
+    // genFnPrologCalleeRegArgs() for argument registers and 
+    // genEnregisterIncomingStackArgs() for stack arguments.
+    genClearStackVec3ArgUpperBits();
+#endif //FEATURE_UNIX_AMD64_STRUCT_PASSING && FEATURE_SIMD
```

Comment:
There is a TODO-64bit comment in genFnPrologCalleeRegArgs() that could be deleted with this change.

Also please update that comment block with the findings of your investigation and what we are doing to have Assumption 2 being valid.